aws/service.ts

```diff
@@ -339,15 +340,13 @@ function computeImageFromBuildWorker(
             };
         });
 
+        buildImageCache.set(imageName, uniqueImageName);
 
-        buildImageCache.set(imageName, imageDigest);
-
-        imageDigest.apply(d =>
+        uniqueImageName.apply(d =>
             pulumi.log.debug(`    build complete: ${imageName} (${d})`, logResource));
     }
 
-    preEnv.IMAGE_DIGEST = imageDigest;
-    return createImageOptions(repositoryUrl, preEnv);
+    return createImageOptions(uniqueImageName, preEnv);
```

Comment:
Note I suspect we no longer need this `preEnv` thing to be wired through here at all?

Reply:
So we technically could remove it.  But we still need to add it to the final object we return.  So this is just easy to keep as is here.  Plus, i don't want to rerun everything :)